src/data/real/cau_seq.lean

```diff
@@ -254,16 +274,24 @@ instance : add_group_with_one (cau_seq β abv) :=
   int_cast_neg_succ_of_nat := λ n, congr_arg const (int.cast_neg_succ_of_nat n),
   .. cau_seq.add_group }
 
+instance : has_pow (cau_seq β abv) ℕ :=
+⟨λ f n, of_eq (npow_rec n f) (λ i, f i ^ n) $ by induction n; simp [*, npow_rec, pow_succ]⟩
+
+-- eligible for `dsimp`
+@[simp, norm_cast, nolint simp_nf]
+lemma pow_apply (f : cau_seq β abv) (n i : ℕ) : (f ^ n) i = f i ^ n := rfl
```

Comment:
I wasn't aware that this type of thing was accepted by norm_cast; but I guess if it weren't lean would complain

Reply:
Yeah, it moves `coe_fn` inward, so it's accepted.